geos: improve dlopen error handling

otan opened the pull request:
* Fix dlopen to return the dlerror messages. dlerror must be called
  before dlclose.

Release justification: low risk, high benefit changes

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/53505)
<!-- Reviewable:end -->

otan committed:
geos: improve dlopen error handling

* Fix dlopen to return the dlerror messages. dlerror must be called
  before dlclose.

Release justification: low risk, high benefit changes

Release note: None

otan commented:
sorry @rmloveland thought my fixed worked but it didn't and i'm stumped.
i'll try again tomorrow.

sumeerbhola reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 3 of 3 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/53505#-:-MFlrAoiDOQlXe9uINu4:bg7igkv)* status: :shipit: complete! 1 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

otan commented:
tftr!

bors r=sumeerbhola
